R/deps.R

```diff
@@ -12,22 +15,17 @@ cran_revdeps <- function(package, dependencies = TRUE, bioc = FALSE) {
 
 #' @importFrom remotes bioc_install_repos
 #' @importFrom crancache available_packages
-
 cran_revdeps_versions <- function(package, dependencies = TRUE, bioc = FALSE) {
   stopifnot(is_string(package))
-  repos <- get_repos(bioc)
 
-  allpkgs <- available_packages(repos = repos)
-  alldeps <- allpkgs[, dependencies, drop = FALSE]
-  alldeps[is.na(alldeps)] <- ""
-  deps <- apply(alldeps, 1, paste, collapse = ",")
-  rd <- grepl(paste0("\\b", package, "\\b"), deps)
-
-  data.frame(
-    stringsAsFactors = FALSE,
-    package = unname(allpkgs[rd, "Package"]),
-    version = unname(allpkgs[rd, "Version"])
+  cache <- pkgcache::cranlike_metadata_cache$new(
+    repos = list(CRAN = "https://cloud.r-project.org"),
```

Comment:
We can use `get_repos(bioc = FALSE)` here. If you hard code the repo, then people cannot override it, and not everyone can use that repo.

It is true that we might pick up extra non-CRAN repos, but that is also the current behavior, and AFAICT there is no real reason to change it.

Reply:
Isn't it worrying that different folks might get different results due to an option?

Reply:
I don't remember that the potential extra repos have ever caused an issue. OTOH people might rely on this. 

I think it is fine either way, but I would not change it at this point.